# HIV-DRLink
HIV-DRLink_github.pl is a script that detects and calculates Linked HIV drug resistant mutations on single HIV genomes. HIV-DRLink is based on processing the DRM information obtained from Stanford HIVdb genotypic resistance interpretation program using the Stanford command line program “Sierra Web Service 2.0: 2016 – present”  (https://hivdb.stanford.edu/page/webservice/). Therefore, the first step is to submit sequences to the Stanford HIVdb. To do this for large scale sequence data, users need to download and install a Python client SierraPy from Stanford HIVdb (https://hivdb.stanford.edu/page/webservice/). The output of SierraPy is a JSON (JavaScript Object Notation) file. Then, HIV-DRlink is used to parse the output file to obtain the drug mutation types and drug resistance mutations (DRM). It then combines all sequences with identical DRM into one sequence and calculates frequencies of linked DRMs and individual DRMs in a sample population.  
The output of HIVdb Sierra JSON files can be extensive, to simplify the output, a simplified GraphQL protocol is used to select only the gene names (PR, RT, or IN), the mutation types (primary type, for example for RT NRTI DRM, NNRTI DRM, or non-drug resistance mutations “other”), and the mutations. The GraphQL protocol used in this pipeline (step 2) is a simple text file called “simple_mutations.gql” 
Step 1: submit sequences to Stanford HIVdb using simple_mutation.gql: sierrapy fasta input_file.fas simple_mutations.gql -o output.json Where input_file is a fasta sequence file and the output.json (or whatever names users prefer) is the output file of HIVdb.
Step 2: Run HIV-DRLink.pl on the output file, output.json, from step1 using the following command line: HIV-DRLink.pl  output.json
